### PR TITLE
Added doc for KIP-535 and updated it for KIP-562

### PR DESCRIPTION
**Pull request by brary**
Documenting KIP-535 in the release notes and making some updates related to KIP-562.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by abbccdda — COMMENTED** [review]
Thanks for the PR, the content looks good. Could you also do a documentation build and screenshot the changes in the summary? Detailed steps: https://cwiki.apache.org/confluence/display/KAFKA/Setup+Kafka+Website+on+Local+Apache+Server

**Comment by brary** [comment]
Hey @abbccdda there seems to be some permissions issue on my mac, tried it for few hours but couldn't make it run after following instructions in the above page.

File: docs/upgrade.html [hunk 1]
```diff
@@ -39,7 +39,8 @@ <h5><a id="upgrade_250_notable" href="#upgrade_250_notable">Notable changes in 2
         <a href="https://github.com/apache/kafka/tree/2.5/examples">examples</a> folder. Check out
         <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-447%3A+Producer+scalability+for+exactly+once+semantics">KIP-447</a>
         for the full details.</li>
-    <li>Deprecated <code>KafkaStreams.store(String, QueryableStoreType)</code> and replaced it with <code>KafkaStreams.store(StoreQueryParameters)</code>.</li>
+    <li>Provided support to query stale stores(for high availability) and the stores belonging to a specific partition by deprecating <code>KafkaStreams.store(String, QueryableStoreType)</code> and replacing it with <code>KafkaStreams.store(StoreQueryParameters)</code>.</li>
```
Comment (vinothchandar) [inline docs/upgrade.html]: break this up into two separate updates so that we also cite the kafkaStreams.keyQueryMetadata() api?

Comment (brary) [reply docs/upgrade.html]: Makes sense, that is also an important api.

Comment (brary) [reply docs/upgrade.html]: Do we need to break up this line to add kafkaStreams.keyQueryMetadata()? That is a separate endpoint that returns key specific information, i.e. partition and hosts containing stale tasks in addition to the active hosts which we were able to fetch earlier.

Comment (brary) [reply docs/upgrade.html]: Something like 
` <li>Added a new public api <code>KafkaStreams.queryMetadataForKey(String, K, Serializer) to get detailed information on the key being queried. It provides information about the partition number where the key resides in addition to hosts containing the active and standby partitions for the key.</code></li>`

Comment (abbccdda) [reply docs/upgrade.html]: image
Let's add a space between `stores(` as `stores (`, and add a period after `<code>KafkaStreams.allLocalStorePartitionLags()</code>`

**Comment by abbccdda** [comment]
image
No worry, I generated that for you :)

**Comment by brary** [comment]
> image
> No worry, I generated that for you :)

Thanks @abbccdda

File: docs/upgrade.html (hunk 1 above)
Comment (brary) [reply docs/upgrade.html]: @vinothchandar can you review this again.

Comment (brary) [reply docs/upgrade.html]: Thanks @abbccdda . Addressed both the comments in the latest commit.

**Comment by mjsax** [comment]
Merged to `trunk` and cherry-picked to `2.5`.

Thanks for the PR @brary! Can you also do a PR against `kafka-site` repository so we can get this updates into the web page right away? Just merging in `kafka` repo does not update the web pag.

**Comment by mjsax** [comment]
@brary, did you see my last comment?

**Comment by brary** [comment]
Sorry, missed your earlier comment @mjsax . Is there any documentation around which files need to be changed in kafka-site.

**Comment by mjsax** [comment]
It's the same file as here from the 2.5 folder: https://github.com/apache/kafka-site/tree/asf-site/25

**Pull request merged**